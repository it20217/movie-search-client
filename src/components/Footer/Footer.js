import { Link } from "react-router-dom";
import styles from "./Footer.module.scss"
import logo from "../../assets/images/dev-logo.png"
function Footer(){
	return(
		<footer className={styles.footer}>
			<div className={styles.logo}>
                <img src={logo} alt="Logo"/>
                <span>Movie&nbsp;app!</span>
			</div>
			<div className={styles.links}>
				<div>
                    <Link to="/" className={styles.link}>Home </Link><br /><br />
                    <span>Start exploring our<br />
                    website here</span>
                </div>
				<div>
                    <Link to="/search" className={styles.link}>Search </Link><br /><br />
                    <span>Search the database<br />
                    for movies of your choice</span>
                </div>
				<div>
                    <Link to="/read-more" className={styles.link}>Read More </Link><br /><br />
                    <span>Find out more about<br />
                    movies, TV shows, actors and more...</span>
                </div>
			</div> 
            <div className={styles.side}>
                <address>
                    <span>Contact Us</span><br /><br />
                    Movie app!<br/>
                    email@email.com<br/>
                    0123456789<br/>
                    @movie_app!
                </address>
			</div>
		</footer>
	)
}

export default Footer;