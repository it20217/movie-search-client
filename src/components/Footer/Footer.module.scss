$main-font: sans-serif;
$main-color: #333;
$font-color: #c0c0c0;
$link-color: #ecdb20;

.footer {
	font-family: $main-font;
	background-color: $main-color;
	color: $font-color;
	width: 90%;
	height: 250px;
	margin: 2% auto;
	display: flex;
	justify-content: space-around;
}
.links {
	height: 100%;
	width: 70%;
	margin: auto;
	display: flex;
	justify-content: space-around;
	align-items: center;
	
	.link {
		color: $link-color;
		text-decoration: none;
		font-weight: bold;
		&:hover{
			color: $font-color;
		}
	}
	span{
		font-size: .85em;
	}
}

.logo {
	width: 80px;
	display: flex;
	padding-left: 4%;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	img {
		width: 100%;
	  }
	span{
		padding-top: 5px;;
	}
}
.side{
	width: 15%;
	margin: auto;
	display: flex;
	align-items: center;
	span{
		font-weight: bold;
		color: white;
	}
}

@media screen and (max-width: 900px){
	.footer{
		width: 100%;
		flex-direction: column-reverse;
		justify-content: center;
		align-items: center;
		padding: 50px 0;
	}
    .links{
        flex-direction: column;
		padding: 40px 0 0;
		span{
			display: none;
    	} 
	}
	.logo {
		display: none;
	}
}