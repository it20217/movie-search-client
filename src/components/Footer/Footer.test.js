import React from 'react';
/** Snapshots rendering */
import renderer from "react-test-renderer";
/** Component */
import Footer from './Footer';
/** React router */
import { BrowserRouter as Router } from 'react-router-dom';

describe("Footer page component", () => {
  it("renders correctly", () => {
    const component = renderer.create(<Router><Footer/></Router>).toJSON();
    expect(component).toMatchSnapshot();
  })
})