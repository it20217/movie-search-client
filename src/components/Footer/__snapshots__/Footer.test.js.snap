// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer page component renders correctly 1`] = `
<footer
  className="wrapper"
>
  <div
    className="footer"
  >
    <div
      className="logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Logo"
          className="pic"
          src="cinema-154392__340.webp"
        />
      </a>
    </div>
    <div
      className="links"
    >
      <div>
        <a
          className="link"
          href="/"
          onClick={[Function]}
        >
          Home 
        </a>
        <br />
        <br />
        <a
          className="link"
          href="/read-more"
          onClick={[Function]}
        >
          Read More 
        </a>
        <br />
        <br />
        <a
          className="link"
          href="/contact-us"
          onClick={[Function]}
        >
          Contact us 
        </a>
        <br />
      </div>
    </div>
    <div
      className="side"
    >
      <address>
        <span>
          Movie app!
        </span>
        <br />
        email@email.com
        <br />
        0123456789
        <br />
        @movie_app!
      </address>
    </div>
  </div>
  <div
    className="project"
  >
    <hr />
    <hr />
    <br />
    Project created by 
    <span>
      Jaroslaw Polowy
    </span>
     and 
    <span>
      Igor Tokarev
    </span>
     Â© 2022 
    <a
      href="https://www.ncirl.ie/"
      rel="noreferrer"
      target="_blank"
    >
      NCI
    </a>
  </div>
</footer>
`;
